.github/workflows/test.yml

```diff
@@ -0,0 +1,36 @@
+name: Tests
+
+on:
+  pull_request:
+    paths-ignore:
+    - 'doc/**'
+  push:
+    branches:
+    - main
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-22.04, macos-12]
+    runs-on: ${{ matrix.os }}
+    if: ${{ github.repository_owner == 'cloudflare' }}
```

Comment:
Is this meant to be a security check? Couldn't someone remove it in their PR?

Also we probably want the ability to run these actions on contributors' PRs, is that possible? I think GitHub usually shows a button that the repo owner has to click to approve running an action if the author isn't trusted.